src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -230,14 +230,17 @@ private[internal] trait GlbLubs {
   def weakLub(tps: List[Type]): Type = (
     if (tps.isEmpty)
       NothingTpe
-    else if (tps forall isNumericValueType)
+    else if (shouldUseNumericLub(tps))
       numericLub(tps)
     else if (tps exists typeHasAnnotations)
       annotationsLub(lub(tps map (_.withoutAnnotations)), tps)
     else
       lub(tps)
   )
 
+  private def shouldUseNumericLub(tps: List[Type]) =
+    tps forall (tp => isNumericValueType(tp.widen))
```

Comment:
I've noted the problem with `isNumericSubtype` over in the [other PR](https://github.com/scala/scala/pull/5310#issuecomment-322352052), your test case illuminates it really well.

I would expect the weak LUB of those aliases to be the same as that of the RHS's.

Given `type A1 = T1; type A2 = T2`, `weakLub(T1, T2) =:= weakLub(A1, A2)` should hold.

Setting literal constant types aside for now, I agree that we should put the `deconst` in `isNumericValueType`.

I'm on the verge of suggesting that we retarget all of this PR to 2.13.x, though. Is the bug fix important enough to warrant the risks on the stable branch?